docs/src/chemiscope-sphinx.rst

```diff
@@ -0,0 +1,74 @@
+.. _chemiscope-sphinx:
+
+Chemiscope in `sphinx`
+======================
+
+The chemiscope python module also provides a `sphinx` extension that can process
+a custom RST directive into an interacting viewer, that can be embedded in the 
+generated HTML documentation.
+
+Configuration
+^^^^^^^^^^^^^
+
+Assuming a project is already configured to use `sphinx` to build its documentation,
+in order to be able to use the chemiscope directive one has to first
+add the `chemiscope.sphinx` extension to the `docs/conf.py` file:
+
+.. code-block:: python
+
+   extensions = ["chemiscope.sphinx"]
+
+Chemiscope directive
+^^^^^^^^^^^^^^^^^^^^
+
+In order to include a chemiscope viewer into a RST documentation page one can use a 
+``chemiscope`` directive. When compiled with ``sphinx-builder``, the directive will
+be translated into an HTML embedding of and interactive widget, that loads a JSON 
+file and displays it. The directive specifies two options:  ``filepath`` - that 
+indicates the path relative to the RST file where the chemiscope JSON is to be found, 
+and  ``mode`` that specifies the type of visualization to be used
+(``default``, ``structure``, or ``map``).
+
+For instance, to show a viewer combining structure and property panels you can 
+include the directive
+ 
+.. code-block:: rst
+
+    .. chemiscope::
+        :filepath: datasets/showcase.json.gz
+        :mode: default 
+
+Once compiled, this will show as this widget
+
+.. chemiscope:: 
+    :filepath: datasets/showcase.json.gz
+    :mode: default 
+
+    
+The ``structure`` mode wil show the structures only
+
+.. chemiscope::
+    :filepath: datasets/showcase.json.gz
+    :mode: structure
+ 
+and the ``map`` mode only the property map
+
+.. chemiscope::
+    :filepath: datasets/showcase.json.gz
+    :mode: map 
+    
+Viewing a documentation locally
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Because of CORS restrictions in most browsers, it is not possible to load the
```

Comment:
is this CORS? I though CORS was about getting stuff from another webserver; while here the issue is more that we don't have such a server

Reply:
I _think_ that because of CORS we can't load the file from disk but right now the issue is that we use absolute paths relative to the server root for the includes so OK we can just say that one must use a server and that's it.